src/sentry/api/endpoints/organization_events_stats.py

```diff
@@ -18,6 +21,45 @@
 
 class OrganizationEventsStatsEndpoint(OrganizationEventsEndpointBase):
     def get(self, request, organization):
+
+        should_fetch_bulk = request.query_params.get("bulk") == "1"
+
+        if should_fetch_bulk:
+            stats_requests = request.query_params.getlist("g")
+
+            collect = []
+
+            for stats_request in stats_requests:
+
+                logging.info("g: %s", stats_request)
+
+                payload = json.loads(stats_request)
+
+                if isinstance(payload, dict):
+
+                    new_query_params = QueryDict("", mutable=True)
+
+                    for key, value in payload.items():
+                        if isinstance(value, list):
+                            new_query_params.setlist(key, value)
+                        else:
+                            new_query_params.setlist(key, [value])
+
+                    logging.info("g: %s", new_query_params.lists())
+
+                    # TODO: is this right?
+                    # request_copy = copy.copy(request)
+                    request.GET = new_query_params
```

Comment:
You shouldn't mutate the request data. You could use the endpoint `convert_args` hooks to splice the parsed parameters into the endpoint's argument list.